pkg/pubsub/public_key_topic.go

```diff
@@ -0,0 +1,188 @@
+package pubsub
```

Comment:
@restevens402 I might kill this in the future because I think we can simplify. I am leaving it for now so I can reuse the code and get this PR in. If you see an opportunity to refactor lets do it while keeping the functionality of `public_key_api_handlers.go` as this has a dependency on this code at the moment.